web/src/components/molecules/Asset/Asset/AssetBody/index.tsx

```diff
@@ -0,0 +1,157 @@
+  handleTypeChange,
+  isModalVisible,
+  handleModalCancel,
+  handleFullScreen,
+}) => {
+  const { fileName, createdAt, createdBy } = asset;
+  const { svgRender, handleCodeSourceClick, handleRenderClick } = useHooks();
+  const formattedCreatedAt = dateTimeFormat(createdAt);
+  const displayUnzipFileList = selectedPreviewType === PreviewType.ZIP;
+  const getViewer = (viewer: Viewer | undefined) => {
+    viewerRef = viewer;
+  };
+  const renderPreview = () => {
+    switch (selectedPreviewType) {
+      case PreviewType.GEO:
+      case PreviewType.ZIP:
+        return (
+          <TilesetPreview
+            viewerProps={{
+              terrainProvider: createWorldTerrain(),
+              navigationHelpButton: false,
+              homeButton: false,
+              projectionPicker: false,
+              sceneModePicker: false,
+              baseLayerPicker: false,
+              fullscreenButton: false,
+              vrButton: false,
+              selectionIndicator: false,
+              timeline: false,
+              animation: false,
+              geocoder: false,
+            }}
+            tilesetProps={{
+              url: url,
+            }}
+            onGetViewer={getViewer}
+          ></TilesetPreview>
+        );
+      case PreviewType.IMAGE:
+        return <Image src={url} alt="asset-preview"></Image>;
+      case PreviewType.SVG:
+        return <SVGPreview url={url} svgRender={svgRender} />;
+      default:
+        return <NoSupportedViewer />;
+    }
+  };
+
+  return (
+    <BodyContainer>
+      <BodyWrapper>
+        <Card
+          title={fileName}
+          toolbar={
+            <PreviewToolbar
+              url={url}
+              selectedPreviewType={selectedPreviewType}
+              isModalVisible={isModalVisible}
+              handleCodeSourceClick={handleCodeSourceClick}
+              handleRenderClick={handleRenderClick}
+              handleFullScreen={handleFullScreen}
+              handleModalCancel={handleModalCancel}
+            />
+          }
+        >
+          {renderPreview()}
+        </Card>
+        {displayUnzipFileList && (
+          <Card title="Unzip File">
+            <UnzipFileList style={{ minHeight: "400px" }}></UnzipFileList>
+          </Card>
+        )}
+        <DownloadButton
+          type="ghost"
+          filename={fileName}
+          url={url}
+          displayDefaultIcon={true}
+        ></DownloadButton>
+      </BodyWrapper>
+      <SideBarWrapper>
+        <SideBarCard title="Asset Type">
+          <PreviewTypeSelect
+            style={{ width: "60%" }}
+            value={selectedPreviewType as PreviewType}
+            onTypeChange={handleTypeChange}
+          />
+        </SideBarCard>
+        <SideBarCard title="Created Time">{formattedCreatedAt}</SideBarCard>
+        <SideBarCard title="Created By">{createdBy?.name}</SideBarCard>
+      </SideBarWrapper>
+    </BodyContainer>
+  );
+};
+
+const BodyContainer = styled("div")`
```

Comment:
Please update all these to `styled.HTMLElement` syntax

Reply:
Understood.